tests/common/include/tests/common/material_interaction.inl

```diff
@@ -327,6 +327,58 @@ TEST(material_interaction, telescope_geometry_energy_loss) {
 
     EXPECT_NEAR(new_var_qop, dvar_qop, 1e-10f);
 
+    /********************************
+     * Test with next_surface_aborter
+     *********************************/
+
+    using alt_actor_chain_t =
+        actor_chain<dtuple, propagation::print_inspector, pathlimit_aborter,
+                    parameter_transporter<transform3>, next_surface_aborter,
+                    interactor_t, parameter_resetter<transform3>>;
+    using alt_propagator_t =
+        propagator<stepper_t, navigator_t, alt_actor_chain_t>;
+
+    bound_track_parameters<transform3> alt_bound_param(1u, bound_vector,
+                                                       bound_cov);
+
+    scalar altE(0);
+
+    unsigned int surface_count = 0;
+    while (true) {
```

Comment:
Why is this needed?

Reply:
To make tracks reach the end of detector because the `propagate` only propagates to the next surface

Reply:
Ok, nvm, I thought only propagating to the next surface was what you were trying to do